tests/functional/test_hostproviders_functions.py

```diff
@@ -11,106 +11,89 @@
 # limitations under the License.
 import json
 import os
-import random
 
 import allure
 import pytest
 from adcm_pytest_plugin import utils
-from adcm_pytest_plugin.docker_utils import DockerWrapper
 from coreapi import exceptions
 from jsonschema import validate
 
 
-# pylint: disable=E0401, W0611, W0621
-from tests.library import errorcodes, steps
+# pylint: disable=E0401, W0611, W0621, W0212
+from tests.library import errorcodes
 
 BUNDLES = os.path.join(os.path.dirname(__file__), "../stack/")
 SCHEMAS = os.path.join(os.path.dirname(__file__), "schemas/")
 
 
-@pytest.fixture()
-def adcm(image, request, adcm_credentials):
-    repo, tag = image
-    dw = DockerWrapper()
-    adcm = dw.run_adcm(image=repo, tag=tag, pull=False)
-    adcm.api.auth(**adcm_credentials)
-    yield adcm
-    adcm.stop()
+def test_load_host_provider(sdk_client_fs):
+    sdk_client_fs.upload_from_fs(BUNDLES + "hostprovider_bundle")
+    with allure.step("Check bundle list"):
+        assert len(sdk_client_fs.bundle_list()) == 1
 
 
-@pytest.fixture()
-def client(adcm):
-    return adcm.api.objects
-
-
-def test_load_host_provider(client):
-    steps.upload_bundle(client, BUNDLES + 'hostprovider_bundle')
-    with allure.step('Check bundle list'):
-        assert client.stack.bundle.list() is not None
-
-
-def test_validate_provider_prototype(client):
-    steps.upload_bundle(client, BUNDLES + 'hostprovider_bundle')
-    with allure.step('Load provider prototype'):
-        provider_prototype = json.loads(json.dumps(client.stack.provider.list()[0]))
+def test_validate_provider_prototype(sdk_client_fs):
+    bundle = sdk_client_fs.upload_from_fs(BUNDLES + "hostprovider_bundle")
+    with allure.step("Load provider prototype"):
+        provider_prototype = json.loads(
+            json.dumps(bundle.provider_prototype()._data)
```

Comment:
why you convert json to str and str to json?

Reply:
I guessed that OrderedDict doesn't casts to object, but it does. Will be updated to the following.
```suggestion
            provider_prototype = bundle.provider_prototype()._data
```